src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -547,6 +561,12 @@ internal bool TypeIsCached(Type type)
             return _classes.ContainsKey(type);
         }
 
+        internal void ClearClasses()
```

Comment:
I believe we should also be clearing the `_converters` cache. If a type has changed, I imagine the implementation of a custom converter for the type might soon follow.